metrics.go

```diff
@@ -16,7 +16,11 @@ type Metric interface {
 
 func publish(message Message) {
 	defer publishWG.Done()
-	statQueue <- message
+	select {
+	case statQueue <- message:
+	default:
+		// Channel is full so we are dropping metric
```

Comment:
Would it make more sense to force publish when the channel is full? --> has an implication of flushing too often during spikes.

Reply:
What do you mean by force publish?

Reply:
Instead of dropping the metrics, can we just publish?

Reply:
No, we should be dropping statsite metrics if they aren't going through.  Metric issues shouldn't cause the service to fail.  Otherwise go routines would backup.